src/main.cpp

```diff
@@ -2080,6 +2081,47 @@ void PartitionCheck(bool (*initialDownloadCheck)(), CCriticalSection& cs, const
     }
 }
 
+Consensus::VersionBitsConditionChecker checkCSV(Consensus::DEPLOYMENT_CSV);
```

Comment:
I'll try another implementation that does that. Separating the cache and the logic probably makes sense.